### PR TITLE
Prevent infinite loop of build.sh script for building nightly images

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
Execution of script `./dockerfiles/build.sh` has caused an infinite loop when running nightly build of Che. It happened, because script didn't fail on error, when it tried to import script by relative path, and then went on to invoke `build.sh` recursivelly.

As a workaround, "set -e" will prevent this infinite loop.
Also add manual workflow trigger to restart the job in this PR

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

File: .github/workflows/nightly-build.yml [hunk 1]
```diff
@@ -45,7 +46,9 @@ jobs:
       run: mvn -B clean install -U -Pintegration
     - name: Build docker images
       run: |
-        ./dockerfiles/build.sh 
+        cd dockerfiles
```
Comment (benoitf) [inline .github/workflows/nightly-build.yml]: let's keep call as it is

File: dockerfiles/build.sh [hunk 2]
```diff
@@ -9,6 +9,8 @@
 #
 # See: https://sipb.mit.edu/doc/safe-shell/
 
+set -e
```
Comment (benoitf) [inline dockerfiles/build.sh]: It's had been fixed in other repositories

```
base_dir=$(cd "$(dirname "$0")"; pwd)
. "${base_dir}"/build.include
```

Comment (benoitf) [inline dockerfiles/build.sh]:
```suggestion
base_dir=$(cd "$(dirname "$0")"; pwd)
. "${base_dir}"/build.include
```

Comment (mkuznyetsov) [reply dockerfiles/build.sh]: `set -e` wouldn't hurt though

Comment (benoitf) [reply dockerfiles/build.sh]: you can keep it, but if we include absolute path, first instruction of build.include is `set -e`

File: .github/workflows/nightly-build.yml [hunk 3]
```diff
@@ -45,7 +46,9 @@ jobs:
...
+        ./build.sh 
+        cd ..
```
Comment (benoitf) [inline .github/workflows/nightly-build.yml]:
```suggestion
./dockerfiles/build.sh 
```

**Comment by mkuznyetsov** [comment]
@benoitf thanks, I've applied base_dir in other places of build.sh where it caused problems

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
@mkuznyetsov thanks !

File: .github/workflows/nightly-build.yml [hunk 4]
```diff
@@ -44,8 +45,8 @@ jobs:
     - name: Build with Maven
       run: mvn -B clean install -U -Pintegration
     - name: Build docker images
-      run: |
-        ./dockerfiles/build.sh 
+      run:
+          ./dockerfiles/build.sh
```
Comment (benoitf) [inline .github/workflows/nightly-build.yml:49]: maybe the `|` is still required ?

Comment (mkuznyetsov) [reply .github/workflows/nightly-build.yml:49]: this is not a multiline anymore, so it should work, just as the command above it

**Pull request merged**